src/renderer/dx/DxRenderer.cpp

```diff
@@ -1689,12 +1691,22 @@ try
     // Calculate positioning of our origin.
     const auto origin = (coord * _fontRenderData->GlyphCell()).to_d2d_point();
 
-    // Create the text layout
-    RETURN_IF_FAILED(_customLayout->Reset());
-    RETURN_IF_FAILED(_customLayout->AppendClusters(clusters));
+    if (_usingSoftFont)
+    {
+        // We need to reset the clipping rect applied by the CustomTextRenderer,
+        // since the soft font will want to set its own clipping rect.
+        RETURN_IF_FAILED(_customRenderer->EndClip(_drawingContext.get()));

```

Comment:
Is this necessary? The only place you push a clip, you've got a scope_exit to pop it. I don't think you're otherwise setting `_clipRect` anywhere, but maybe I'm mistaken. Maybe we need this just to make sure there's not _already_ a clip rect